SimPEG/regularization/vector.py

```diff
@@ -421,12 +421,10 @@ def deriv(self, m) -> np.ndarray:
         (n_param, ) numpy.ndarray
             Gradient of the regularization function evaluated for the model provided.
         """
-        d_m = self._delta_m(m)
+        d_m = self.mapping * self._delta_m(m)
```

Comment:
This is a comment for a later pr, but it would be nice to have the option to apply the mapping to the reference model or not. In the amplitude regularization, this would be nice because we would then not have to provide a reference model in cartesian coordinates, we could simply provide a reference amplitude. The derivatives for this would be okay for the smallness term (but would get ugly in smoothness terms. But we could just not allow a reference amplitude in the smoothness). We can create an issue on this to follow up later